web-server/src/content/Dashboards/ConfigureGitlabModalBody.tsx

```diff
@@ -0,0 +1,345 @@
+                  return;
+                }
+              }}
+              error={!!showDomainError.value}
+              sx={{ width: '100%' }}
+              value={customDomain.value}
+              onChange={(e) => handleDomainChange(e.currentTarget.value)}
+              label={
+                isDomainInputFocus.value || customDomain.value
+                  ? 'Custom Domain'
+                  : '(Optional)'
+              }
+              onFocus={isDomainInputFocus.true}
+              onBlur={isDomainInputFocus.false}
+            />
+          </FlexBox>
+          <Line error tiny mt={1} minHeight={'18px'}>
+            {showDomainError.value}
+          </Line>
+          <FlexBox>
+            <Line tiny mt={1} primary sx={{ cursor: 'pointer' }}>
+              <Link
+                href="https://gitlab.com/-/user_settings/personal_access_tokens"
+                target="_blank"
+                rel="noopener noreferrer"
+              >
+                <Line
+                  underline
+                  sx={{
+                    textUnderlineOffset: '2px'
+                  }}
+                >
+                  Generate new classic token
+                </Line>
+              </Link>
+              <Line ml={'5px'}>{' ->'}</Line>
+            </Line>
+          </FlexBox>
+        </FlexBox>
+
+        <FlexBox justifyBetween alignCenter mt={'auto'}>
+          <FlexBox col sx={{ opacity: 0.8 }}>
+            <Line>Learn more about Gitlab</Line>
+            <Line>
+              Personal Access Token (PAT)
+              <Link
+                ml={1 / 2}
+                href="https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html"
+                target="_blank"
+                rel="noopener noreferrer"
+              >
+                here
+              </Link>
+            </Line>
+          </FlexBox>
+          <FlexBox gap={2} justifyEnd>
+            <LoadingButton
+              loading={isLoading.value}
+              variant="contained"
+              onClick={handleSubmission}
+            >
+              Confirm
+            </LoadingButton>
+          </FlexBox>
+        </FlexBox>
+      </FlexBox>
+      <Divider orientation="vertical" flexItem />
+      <TokenPermissions />
+    </FlexBox>
+  );
+};
+
+const TokenPermissions = () => {
+  const imageLoaded = useBoolState(false);
+
+  const expandedStyles = useMemo(() => {
+    const baseStyles = {
+      border: `2px solid ${alpha('rgb(256,0,0)', 0.6)}`,
+      transition: 'all 0.8s ease',
+      borderRadius: '8px',
+      opacity: 1,
+      width: '126px',
+      position: 'absolute',
+      maxWidth: 'calc(100% - 48px)',
+      left: '12px'
+    };
+
+    return [
+      {
+        height: '32px',
+        top: '298px'
+      },
+      {
+        height: '32px',
+        top: '360px'
+      },
+      {
+        height: '32px',
+        top: '425px'
+      }
+    ].map((item) => ({ ...item, ...baseStyles }));
+  }, []);
```

Comment:
Please explain this

Reply:
the squares that highlight the necessary permissions

<img width="178" alt="image" src="https://github.com/user-attachments/assets/f73a1886-a05e-48b4-9c96-96674baf1c1e">

Reply:
why we need css for that? are we expanding the images when user hover the mouse on the image

Reply:
Approving it anyhow, but why not we are simply adding the red boxes in the image itself .